store/v2/root/cache_store.go

```diff
@@ -0,0 +1,37 @@
+package root
+
+import (
+	"github.com/cosmos/cosmos-sdk/store/cachekv"
+	types "github.com/cosmos/cosmos-sdk/store/v2"
+)
+
+// GetKVStore implements BasicMultiStore.
+func (cs *cacheStore) GetKVStore(skey types.StoreKey) types.KVStore {
+	key := skey.Name()
+	sub, has := cs.substores[key]
+	if !has {
+		sub = cachekv.NewStore(cs.source.GetKVStore(skey))
+		cs.substores[key] = sub
+	}
+	// Wrap with trace/listen if needed. Note: we don't cache this, so users must get a new substore after
+	// modifying tracers/listeners.
+	return cs.wrapTraceListen(sub, skey)
+}
+
+// Write implements CacheMultiStore.
+func (cs *cacheStore) Write() {
+	for skey, sub := range cs.substores {
+		sub.Write()
+		delete(cs.substores, skey)
```

Comment:
Why are we deleting the key here?

Reply:
I was thinking that cache stores are invalidated after writing, so the substores can be cleared. But I actually misunderstood this, it seems they can be reused. So  we shouldn't delete here